native/core/src/execution/shuffle/shuffle_writer.rs

```diff
@@ -1570,13 +1588,41 @@ pub fn write_ipc_compressed<W: Write + Seek>(
     // write ipc_length placeholder
     output.write_all(&[0u8; 8])?;
```

Comment:
I know this isn't your code, but it jumped out at me: can we replace this 8-byte write with a seek?
```
// seek past ipc_length placeholder
    output.seek_relative(8)?;
```